java/arcs/core/storage/StorageProxy.kt

```diff
@@ -127,6 +127,14 @@ class StorageProxy<Data : CrdtData, Op : CrdtOperationAtTime, T>(
         busySendingMessagesChannel.asFlow().debounce(50).filter { !it }.first()
     }
 
+    /**
+     * Suspends until there are no more outgoing messages or handle notifications in flight.
+     */
+    suspend fun waitForIdle() {
+        awaitOutgoingMessageQueueDrain()
+        scheduler.waitForIdle()
```

Comment:
Could the outgoing message drain enqueue more scheduler work? :(

Reply:
no, the outgoing message drain wouldn't do that directly - but I suppose it's possible for another handle on another proxy to react to the update coming from one of the outgoing messages by performing an action that would cause another `onUpdate` to need to be handled on this storage proxy's scheduler

Reply:
I've swapped the order but didn't add a timeout.